bsm_generator/src/bsm_generator_node.cpp

```diff
@@ -181,9 +179,12 @@ namespace bsm_generator
     bsm_.core_data.longitude = coord.lon;
     bsm_.core_data.latitude = coord.lat;
     bsm_.core_data.elev = coord.ele;
+    
+    bsm_.core_data.heading = worker->getHeadingInRange(static_cast<float>(worker->getHeading(msg->pose.orientation)));
```

Comment:
Is this static_cast necessary? It looks like this function already returns a float, and it does a static_cast in it's return statement even.

Reply:
It is not necessary. There are some small sonar code smells I could address here including this one if we want?